example/example-client.ts

```diff
@@ -0,0 +1,24 @@
+import type { Socket as SocketBase } from "socket.io-client";
+
+export namespace Root {
+  export interface Emission {
+    time: (currentIsoTime: string) => void;
+    chat: (
+      message: string,
+      extraInfo: {
+        /** the author ID */
+        from: string;
+      },
+    ) => void;
+    rooms: (roomIds: string[]) => void;
+  }
+  export interface Actions {
+    chat: (message: string) => void;
+    ping: (
+      cb1: (literally: "pong", ...theEchoOfTheInput: unknown[]) => void,
```

Comment:
that revealed that the payload schema can not have both `rest` and acknowledgement at the same time, because `...` can not appear in the middle.
There are some hacks for that but that do not work well.

Reply:
That one actually works:
https://www.typescriptlang.org/play?#code/C4TwDgpgBAaghgJwMJwDaoEZwMYGsA8AKlBAB7AQB2AJgM5RyUgDaAugHxQC8UAFAHSDEAc1oAuKM0H9CAGj4QECAPwSAokoD2CAJTdOAN00BLaqz1dDJ6gCgboSFACyIFOix5usRG8w4CzLTACMaUwvKUAK4AthiKHDYA9Il8cACMEkEhYfJwAEwSUbGKuQDMEryKKupaCFAAPlCRNBAAZqEQ1BZWpt1QRqZ22JqUQVDRrhgSLr4euF68tBHy2Bh9AN6JAFRQIJqRdcbRYKgQ0VTAcMDGI1BbKQC+SSlLlLKrDAjQWSPCqCBQBydKCUTQAdyAA

Reply:
But it implies the fixed tuple for rest, known number of arguments — it's useless complication

Reply:
This one works:

![image](https://github.com/RobinTail/zod-sockets/assets/13189514/f5a85b08-7d25-4da4-99e5-e75bb8048197)

Reply:
fixed by 5eaf9fd